test/unit/actions/preupgrade_action_test.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require 'test_plugin_helper'
+
+module ForemanLeapp
+  class PreupgradeActionTest < ActiveSupport::TestCase
+    include Dynflow::Testing
+
+    let(:rex_feature) { RemoteExecutionFeature.find_by(label: 'leapp_preupgrade') }
+    let(:klass) { Actions::ForemanLeapp::PreUpgradeAction }
+
+    describe 'plan' do
+      test 'run plan phase' do
+        job_invocation = FactoryBot.create(:job_invocation, :with_task, :with_template,
+                                           job_category: ::ForemanLeapp::JOB_CATEGORY,
+                                           remote_execution_feature: rex_feature)
+        template_invocation = job_invocation.template_invocations.first
+        host = template_invocation.host
+        planned_action = create_and_plan_action(klass, job_invocation, host, template_invocation)
+
+        assert_equal planned_action.input['host_id'], host.id
+        assert_equal planned_action.input['job_invocation_id'], job_invocation.id
```

Comment:
It would be also good to assert the planned action has a step in finalize here, see https://github.com/Dynflow/dynflow/blob/master/lib/dynflow/testing/assertions.rb#L65